### PR TITLE
Generalize remove_if to eliminate_dead_code

### PR DESCRIPTION
**Pull request by SamirDroubi**
* Implement dead loop elimination
* This commit renames a scheduling operation, so it isn't backwards compatible

### CODE REVIEW

**Comment by SamirDroubi** [comment]
I originally implemented this because I wanted to be able to add guards around loops with the condition being the lo < hi. Plan was to use specialize to add a guard around the loop, then use dead code elimination to remove the loop from the else branch. However, I just relaized that specialize could never forward a statement. So, I am not sure if I should merge this now.

**Comment by SamirDroubi** [comment]
I think specialize is another example where there is no correct place to forward a statement to, but rather you might want to return a list of relevant cursors.

**Comment by yamaguchi1024** [comment]
@SamirDroubi Is this ready to review?

**Comment by SamirDroubi** [comment]
@yamaguchi1024 yes!

File: tests/golden/test_schedules/test_eliminate_dead_code7.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def foo(n: size):
+    if 0 < n:
+        for i in seq(0, n):
+            x: f32 @ DRAM
+    else:
+        pass
```
[the remark this answers is not in the document]
Comment (yamaguchi1024) [reply tests/golden/test_schedules/test_eliminate_dead_code7.txt:6]: I think you'd want a test with multiple loops in the scope

**Review by yamaguchi1024 — APPROVED** [review]

**Pull request merged**